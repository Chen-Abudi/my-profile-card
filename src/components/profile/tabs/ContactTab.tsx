import React from "react";

const ContactTab: React.FC = () => {
  return (
    <div>
      <h2 className="text-xl font-bold text-blue-700">Contact</h2>
      <p>
        If you have any questions or would like to collaborate on a project,
        please feel free to reach out to me. You can find me on various social
        media platforms, including GitHub, LinkedIn, and Medium. Additionally,
        you can email me at{" "}
        <a
          href="mailto:cheezumcali23@gmail.com"
          className="font-medium text-blue-900 hover:text-blue-500"
        >
          GMail
        </a>
        .
      </p>
    </div>
  );
};

export default ContactTab;
