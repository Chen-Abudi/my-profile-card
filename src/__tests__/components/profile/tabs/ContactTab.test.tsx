import React from "react";
import { render, RenderResult } from "@testing-library/react";
import "@testing-library/jest-dom";

import ContactTab from "@/components/profile/tabs/ContactTab";

describe("ContactTab", () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(<ContactTab />);
  });

  it("should render the Contact section", () => {
    const { getByText } = component;

    expect(getByText("Contact")).toBeInTheDocument();
  });

  it("should render the contact paragraph", () => {
    const { getByText } = component;

    expect(
      getByText(
        /If you have any questions or would like to collaborate on a project/
      )
    ).toBeInTheDocument();
  });

  it("should render the email link with the correct href", () => {
    const { getByText } = component;
    const emailLink = getByText("GMail");

    expect(emailLink).toBeInTheDocument();
    expect(emailLink).toHaveAttribute("href", "mailto:cheezumcali23@gmail.com");
  });

  it("should mention social media platforms", () => {
    const { getByText } = component;

    expect(getByText(/GitHub/)).toBeInTheDocument();
    expect(getByText(/LinkedIn/)).toBeInTheDocument();
    expect(getByText(/Medium/)).toBeInTheDocument();
  });

  it("should apply the correct CSS classes", () => {
    const { getByText } = component;
    const heading = getByText("Contact");

    expect(heading).toHaveClass("text-xl", "font-bold", "text-blue-700");

    const emailLink = getByText("GMail");
    expect(emailLink).toHaveClass(
      "font-medium",
      "text-blue-900",
      "hover:text-blue-500"
    );
  });

  it("should match the snapshot", () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
  });
});
