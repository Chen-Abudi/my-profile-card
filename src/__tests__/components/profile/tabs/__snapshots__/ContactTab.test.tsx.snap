// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTab should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-xl font-bold text-blue-700"
    >
      Contact
    </h2>
    <p>
      If you have any questions or would like to collaborate on a project, please feel free to reach out to me. You can find me on various social media platforms, including GitHub, LinkedIn, and Medium. Additionally, you can email me at 
      <a
        class="font-medium text-blue-900 hover:text-blue-500"
        href="mailto:cheezumcali23@gmail.com"
      >
        GMail
      </a>
      .
    </p>
  </div>
</DocumentFragment>
`;
